### ISSUE TITLE
Calculator Bug: Incorrect computation of powers of 2 larger than 54

### ISSUE THREAD

**Issue by ross-pure** [duckduckgo/zeroclickinfo-goodies#4722]
### Description
Computing `2^x` when `x` is greater than `54` gives incorrect results. 

## Steps to recreate
Input `2^x` into the calculator for some `x > 54`. For example, the result for `x = 63` is given in the link https://duckduckgo.com/?q=2%5E63&t=ffab&ia=calculator and is wrong (larger by `192`).

## Additional Information
If we call the result of the calculator `y`, then looking at the difference `d = y - 2^x` is suggestive; the following list gives `d` in binary for various `x`:

- `x = 55`: `10`
- `x = 56`: `100`
- `x = 57`: `-10`
- `x = 58`: `-100`
- `x = 59`: `1100`
- `x = 60`: `11000`
- `x = 61`: `110000`
- `x = 62`: `1100000`
- `x = 63`: `11000000`

---

Instant Answer Page: https://duck.co/ia/view/calculator

**Comment by moollaza** [comment]
@ross-pure thanks for reporting. I think this is probably due to a rounding error? These numbers are pretty big.

**Comment by ross-pure** [comment]
@moollaza Yeah I'm not familiar with perl/js so a quick browse of the codebase didn't reveal much to me.

On the one hand, it makes sense that a search engine calculator doesn't need to support large number calculations, but on the other hand I often google these sort of computations (e.g. powers of 2) when I'm debugging my own code, and the incorrect result sent me down a (however brief) rabbit hole, so I thought I should report it anyway.

It seems that the google calculator gives not many significant figures for these larger powers of `2`. Naturally this could be seen as bad since it is not as accurate, but on the other hand when I see such a result I know I need to go use a proper calculator; seeing the precision of the duckduckgo result tricked me into thinking it was exact. So it's a trade off - personally I don't feel particularly strongly either way.

**Comment by unknownSDE** [comment]
what if handle large numbers correctly or display proper warning like -  number is big to show the result try with some small numbers?
